src/main/kotlin/app/SourcererApi.kt

```diff
@@ -25,74 +26,70 @@ object SourcererApi {
-        val (request, response, result) = Fuel.get("/commit")
-                .authenticate(username, password)
+    private fun createRequestPostCommit(commit: Commit): Request {
+        return Fuel.post("/commit").authenticate(username, password)
                 .body(commit.serialize())
-                .responseString()
+    }
+
+    private fun <T> makeBlockingRequest(request: Request, requestName: String,
+                                parser: (String) -> T): T {
+        try {
+            Logger.debug("Request $requestName initialized")
+            val (_, _, result) = request.responseString()
+            val body = result.get()
+            Logger.debug("Request $requestName success")
+            return parser(body)
+        } catch (e: FuelError) {
+            Logger.error("Request $requestName error", e)
+            throw RequestException(e)
+        }
 
-        val body = result.get()
-        Logger.debug("Request $name success")
-        return body
     }
 
-    fun getUserAsync(success: (String) -> Unit, failure: (String) -> Unit) {
-        val name = "getUserAsync"
-        Logger.debug("Request $name initialized")
-        Fuel.get("/user")
-                .authenticate(username, password)
-                .responseString { request, response, result ->
-                    when (result) {
-                        is Result.Failure -> {
-                            Logger.error("Request $name error",
-                                    result.getException())
-                            failure(result.get())
-                        }
-                        is Result.Success -> {
-                            Logger.debug("Request $name success")
-                            Logger.debug(result.get())
-                            success(result.get())
-                        }
-                    }
+    private fun makeAsyncRequest(request: Request, requestName: String,
+                                 success: (String) -> Unit,
+                                 failure: (String) -> Unit) {
+        Logger.debug("Request $requestName initialized")
+        request.responseString { _, _, result ->
+            when (result) {
+                is Result.Failure -> {
+                    Logger.error("Request $requestName error",
+                            result.getException())
+                    failure(result.get())
+                }
+                is Result.Success -> {
+                    Logger.debug("Request $requestName success")
+                    Logger.debug(result.get())
+                    success(result.get())
                 }
+            }
+        }
+        Logger.debug("Request $requestName success")
+    }
+
+    fun getUserBlocking(): User {
+        return makeBlockingRequest(createRequestGetUser(),
+                "getUserBlocking",
+                { body -> User().parseFrom(body) })
+    }
+
+    fun getUserAsync(success: (String) -> Unit, failure: (String) -> Unit) {
```

Comment:
The reason here is to make a more clear description of the behaviour of functions.
For example, getUserBlocking used in UI while blocking auth check. postCommitAsync used in repo hashing to send prepared data while processing remaining commits.